docs/extensions.md

```diff
@@ -0,0 +1,365 @@
+
+```
+node index.js localhost:23855 65f6b964-a19f-4698-b2db-4e171420439b
+```
+
+Requests that do not provide the auth token will result in 401 Unauthorized responses.
+
+### Endpoint reference
+
+* [`Command`](#command)
+* [`Out`](#out)
+* [`Info`](#info)
+* [`Warn`](#warn)
+* [`Error`](#error)
+* [`Prompt`](#prompt)
+* [`Query`](#query)
+* [`Fail`](#fail)
+
+
+#### `Command`
+
+Used to collect user context for the current command execution.  Contains the command to be run as well as any flag values supplied by the user.  This is the first call the extension should make after initializing.
+
+**URL** : `/command`
+
+**Method** : `GET`
+
+**Response**
+
+```json
+{
+    "command": "hello",
+    "flags": {
+      "name": "Shelly"
+    }
+}
+```
+
+#### `Out`
+
+Used to send output to the user's terminal.
+
+**URL** : `/out`
+
+**Method** : `POST`
+
+**Request body**
+```json
+"Hello Shelly!"
+```
+
+
+#### `Info`
+
+Used to send an informational log message to the user's terminal.
+
+**URL** : `/info`
+
+**Method** : `POST`
+
+**Request body**
+```json
+"Generating hello message..."
+```
+
+#### `Warn`
+
+Used to send a warning log message to the user's terminal.
+
+**URL** : `/warn`
+
+**Method** : `POST`
+
+**Request body**
+```json
+"Hello message generation is taking longer than usual."
+```
+
+#### `Error`
+
+Used to send an error log message to the user's terminal.
+
+**URL** : `/error`
+
+**Method** : `POST`
+
+**Request body**
+```json
+"Could not generate hello message."
+```
+
+#### `Prompt`
+
+Used to collect user input via an interactive prompt.  `options`, `default`, and `placeholder` can be provided but are not required.
+
+**URL** : `/prompt`
+
+**Method** : `POST`
+
+**Request body**
+```json
+{
+  "prompt": "Who would you like to say hello to?",
+  "options": ["Shelly", "Tim", "Ramses"],
+  "default": "Shelly",
+  "placeholder": "Shelly",
+}
+```
+
+**Response**
+
+```json
+"Shelly"
+```
+
+#### `Query`
+
+Used to execute an authenticated request against the NerdGraph API.  The host CLI will use the current user's CLI profile to determine how to authenticate to the NerdGraph backend.
+
+**URL** : `/query`
+
+**Method** : `POST`
+
+**Request body**
+```json
+"{ actor { user { name } } }"
+```
+
+**Response**
+
+```json
+{
+  "data": {
+    "actor": {
+      "user": {
+        "name": "Shelly"
+      }
+    }
+  }
+}
+```
+
+#### `Fail`
+
+Used to indicate to the host CLI that the extension has failed.
+
+**URL** : `/fail`
+
+**Method** : `POST`
+
+## Publishing and installing extensions
+To allow the CLI host to install the extension, it must be publicly available via GitHub.  It must also include an `extension.yml` in the project root as defined above.
+
+Users can then install a plugin via the `newrelic plugins add` command:
```

Comment:
```diff
- newrelic plugins add
- newrelic extensions add ?
```

Reply:
Just want to be consistent for when we present this 🙂

Reply:
👍